src/app/components1/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit } from '@angular/core';
+import Product from '../Types/Product';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ProductService } from '../product.service';
+import { AuthentificationService } from '../authentification.service';
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss']
+})
+export class ProductDetailComponent implements OnInit {
+
+  constructor(private productService: ProductService, private route: ActivatedRoute,private router: Router,private authService: AuthentificationService) {
+  }
+
+  product: Product = {
+    id: 0,
+    description: '',
+    name: '',
+    price: 0,
+    category: '',
+    image: '',
+  };
+
+  canEditProduct = this.authService.userHasRole('admin');
```

Comment:
**suggestion:** you should have this assignment in the component lifecycle, more specifically in `ngOnInit`